### ISSUE TITLE
maxReplicaCount doesnt work for Scaling Jobs using ScaledObject 

### ISSUE THREAD

**Issue by dangelm** [kedacore/keda#749]
i am trying to execute a job per message in rabbitmq.
Setting maxReplicaCount for ScaledObject of 'job' type is not doing anything. i would expect that if i set this value to x, and there is x messages in the queue, x jobs will be created.
my yaml file looks like this:

apiVersion: keda.k8s.io/v1alpha1
kind: ScaledObject
metadata:
  name: {scaled-object-name}
spec:
  scaleType: job
  jobTargetRef:
	deploymentName: {scaled-object-name}
    parallelism: 1 
    completions: 1 
    activeDeadlineSeconds: 300 
    backoffLimit: 1
    template:
      # job description - starting image with nodejs app with rabbit listener
  pollingInterval: 5  
  cooldownPeriod:  5 
  minReplicaCount: 0   
  maxReplicaCount: 20 
  triggers:
    - type: prometheus
      metadata:
        serverAddress: {promotheus address}
        metricName: {metric name for rabbit metrics in promotheus}
        threshold: 1
        query: {queue-name}

## Expected Behavior
for x messages in the queue i would expect x jobs would be created.

## Actual Behavior
only 1 job is created and only after termination a new job spawns.

## Steps to Reproduce the Problem

  1. start 5 jobs simultaneously

## Specifications

  - **KEDA Version:** 1.3
  - **Platform & Version:** linux 64bit
  - **Kubernetes Version:** Major:"1", Minor:"15", GitVersion:"v1.15.10"
  - **Scaler(s):** Prometheus

**Comment by asipras** [comment]
This is similar to what I have noticed when trying to scale jobs using Azure Service Bus trigger. I had opened support issue #652. At this point scale jobs feature using KEDA is not usable.

**Comment by tomkerkhove** [comment]
Closing in favor of #801
